From 24ddd5a7c1ced820179c171421495acaa1fe1ad9 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Mon, 17 Jan 2022 16:04:32 +0800
Subject: [PATCH] net/ramaxel: fix compile issue of spnic_ethtool.c
Patch-mainline: Never, fixes out-of-tree driver
References: patch dependencies

Since commit 7462494408cd3 ("ethtool: extend ringparam setting/getting
API with rx_buf_len") addes two new parameters kernel_ringparam and
extack for .get_ringparam and .set_ringparam.

Let's update spnic driver to reflect the change.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/ramaxel/spnic/spnic_ethtool.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool.c b/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool.c
index dc49395c47d5..e3a7e81a601b 100644
--- a/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool.c
+++ b/drivers/net/ethernet/ramaxel/spnic/spnic_ethtool.c
@@ -94,7 +94,9 @@ int spnic_nway_reset(struct net_device *netdev)
 	return 0;
 }
 
-static void spnic_get_ringparam(struct net_device *netdev, struct ethtool_ringparam *ring)
+static void spnic_get_ringparam(struct net_device *netdev, struct ethtool_ringparam *ring,
+				struct kernel_ethtool_ringparam *kernel_ering,
+				struct netlink_ext_ack *extack)
 {
 	struct spnic_nic_dev *nic_dev = netdev_priv(netdev);
 
@@ -141,7 +143,9 @@ static int check_ringparam_valid(struct net_device *netdev, struct ethtool_ringp
 	return 0;
 }
 
-static int spnic_set_ringparam(struct net_device *netdev, struct ethtool_ringparam *ring)
+static int spnic_set_ringparam(struct net_device *netdev, struct ethtool_ringparam *ring,
+			       struct kernel_ethtool_ringparam *kernel_ering,
+			       struct netlink_ext_ack *extack)
 {
 	struct spnic_nic_dev *nic_dev = netdev_priv(netdev);
 	struct spnic_dyna_txrxq_params q_params = {0};
-- 
2.26.2

